### PR TITLE
1.12/1652 fix the auto generation of voucher numbers vv e

### PR DESCRIPTION
**Pull request by charlesstrange2**
https://trello.com/c/uWQwSt71/1652-fix-the-auto-generation-of-voucher-numbers-vv-e

### CODE REVIEW

File: resources/views/service/vouchers/create.blade.php [hunk 1]
```diff
@@ -1,48 +1,77 @@
...
-                    @if ($errors->has('start')) <label for="start" class="alert-danger">{{ $errors->first('start') }}</label> @endif
-                    @if ($errors->has('start-serial')) <label for="start" class="alert-danger">{{ $errors->first('start-serial') }}</label> @endif
-                </div>
+            <h1>Add a batch of voucher codes</h1>
+
+            <p>Use the form below to add a new batch of vouchers. Select an area, and then enter the starting and ending
+                voucher code numbers.</p>
+
+            <form role="form" method="POST" action="{{ route('admin.vouchers.storebatch') }}" class="styled-form">
+                {!! csrf_field() !!}
```
Comment (AlinChindea) [inline resources/views/service/vouchers/create.blade.php:14]: I don't think we need to escape it here or if we must, I'd use the @csrf directive instead 👀

File: tests/Unit/Controllers/Service/Admin/VoucherControllerTest.php [hunk 2]
```diff
@@ -114,4 +114,29 @@ public function testStoreBatch() {
...
+        // Assert that 3 vouchers have been made and are in the expected state and land within the expected range.
+        $this->assertCount(3, $vouchers);
+        foreach ($vouchers as $voucher) {
+            $this->assertEquals('printed', $voucher->currentstate);
+
+            // Assert that the voucher code starts with the Sponsor shortcode.
+            $this->assertStringStartsWith($shortcode, $voucher->code);
+
+            // the voucher code must be padded to 5
+            $this->assertRegExp('/'. $shortcode . '(09999|10000|10001)$/', $voucher->code);
```
Comment (AlinChindea) [inline tests/Unit/Controllers/Service/Admin/VoucherControllerTest.php:139]:
```suggestion
            $this->assertRegExp('/' . $shortcode . '(09999|10000|10001)$/', $voucher->code);
```

**Review by AlinChindea — APPROVED** [review]
🐸 all good - just a q and a minor suggestion but can live with those.

File: resources/views/service/vouchers/create.blade.php (hunk 1 above)
Comment (charlesstrange2) [reply resources/views/service/vouchers/create.blade.php:14]: oh, that appears to have come in in 5.6... and we missed it. neat.
I'll make a card to modernise our blade

**Pull request merged**